src/controllers/info.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from "express";
+import { getHostname } from "../config";
+
+export function idGet(req: Request, res: Response) {
+  res.json(buildClientIdDocument())
+}
+
+export function buildClientIdDocument(): ClientIdDocument {
+ return {
+    "@context": [
+      "https://www.w3.org/ns/solid/oidc-context.jsonld"
+    ],
+    "client_id": `${getHostname()}/info/id`,
+    "client_name": "GDS Solid proof of concept app",
+    "client_uri": `${getHostname()}`,
+    "post_logout_redirect_uris": [
+      `${getHostname()}`
+    ],
+    "redirect_uris": [
+      `${getHostname()}/login/callback`
+    ]
+  }
+}
+
+export type ClientIdDocument = {
```

Comment:
Yeah, we shouldn't be making you create this type here, but for now you do need to (so fair play to you!). We've created an internal ticket to expose this type for you from `solid-client-authn-js` (just for future reference, so this is good feedback for us!).

Reply:
Ah nice, I didn't even think to look in the client libraries for this 😅